elixirkit/lib/elixirkit/nif.ex

```diff
@@ -0,0 +1,12 @@
+defmodule ElixirKit.NIF do
+  @moduledoc false
+  @on_load {:__init__, 0}
+
+  def __init__ do
+    :erlang.load_nif(Application.app_dir(:elixirkit, "priv/elixirkit_nif"), 0)
```

Comment:
FWIW, you could also have different directory+files per OS, in case it makes it easier to manage than a single large NIF.

Reply:
good call!